### PR TITLE
[DEVEX-152] Add CI workflow to validate changes in Terraform code

### PR DESCRIPTION
**Pull request by Krusty93**
### List of changes

Add CI workflow to run `pre-commit` hook at PR creation/modification time
Validation is performed only on modified files

### Motivation and context

Be sure to validate each Terraform change (formatting, security, etc.) to do not break the code.

### Type of changes

- [X] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### CODE REVIEW

File: .github/workflows/static_analysis.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          git fetch --no-tags --prune --depth=1 origin +refs/heads/*:refs/remotes/origin/*
+
+      - name: Get changed files
+        id: file_changes
+        run: |
+          set -eu
+
+          export DIFF=$(git diff --name-only origin/${{ github.base_ref }} ${{ github.sha }})
+          echo "Diff between ${{ github.base_ref }} and ${{ github.sha }}"
+          echo "files=$( echo "$DIFF" | xargs echo )" >> $GITHUB_OUTPUT
```
Comment (gunzip) [inline .github/workflows/static_analysis.yaml:47]: shouldn't we escape `$DIFF` someway?

File: .github/workflows/static_analysis.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+name: Terraform Analysis
```
Comment (gunzip) [inline .github/workflows/static_analysis.yaml:1]: can we have a comment / description about what this action is supposed to do?

**Review by gunzip — CHANGES_REQUESTED** [review]

File: .github/workflows/static_analysis.yaml (hunk 1 above)
Comment (Krusty93) [reply .github/workflows/static_analysis.yaml]: Added

File: .github/workflows/static_analysis.yaml (hunk 2 above)
Comment (Krusty93) [reply .github/workflows/static_analysis.yaml]: Added

File: .github/workflows/static_analysis.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
+# Perform validation using pre-commit hook on .tf files
```
Comment (gunzip) [inline .github/workflows/static_analysis.yaml:1]:
```suggestion
# Perform validation using pre-commit hook on .tf files
```
```suggestion
# This pipeline assumes the presence of a .pre-commit-config.yaml file at 
# the top level of the repository, which contains Terraform validation tasks.
# For more details, refer to the pre-commit configuration file:
# https://github.com/pagopa/dx/blob/main/.pre-commit-config.yaml.
```

**Review by gunzip — APPROVED** [review]

**Pull request merged**